homeworks/08-functional-reactive-programming/src/renderer.ts

```diff
@@ -0,0 +1,73 @@
+import window from "./assets/window.png";
+import wall from "./assets/wall.png";
+import happyCat from "./assets/happy_cat.png";
+import darkCat from "./assets/cat_dark.png";
+import beigeCat from "./assets/cat_beige.png";
+import dottedCat from "./assets/cat_dots.png";
+
+const canvas = <HTMLCanvasElement>document.getElementById("canvas");
+const ctx = canvas.getContext("2d")!;
+export const generateBtn = document.getElementById("generate")!;
+export const score = document.getElementById("score")!;
+
+const startY = 50;
+const NUMBER_OF_ROWS = 10;
+const images: string[] = [happyCat, darkCat, beigeCat, dottedCat];
+const random = (): number => Math.floor(Math.random() * NUMBER_OF_ROWS);
+
+const drawInitialImg = (image: string, coords: number[]): void => {
+  const img = new Image();
+  img.onload = () => {
+    ctx.drawImage(img, coords[0], coords[1] + 10);
+  };
+  img.src = image;
+};
+
+export const genField = (): number[][] => {
```

Comment:
Please split this function into several functions: 
- getting window positions
- drawing

Reply:
Split the drawing function. Moved the drawing to the main stream flow.
Commit [991e601](https://github.com/sefrem/evo-ts-bootcamp/pull/7/commits/991e6011d928f85d518b41b2099fdb1ca1bcb1bb)